### PR TITLE
Don't try to insert queue records if no downstream dag

### PR DESCRIPTION
**Pull request by dstandish**

### CODE REVIEW

File: airflow/datasets/manager.py [hunk 1]
```diff
@@ -96,12 +96,11 @@ def _slow_path_queue_dagruns(self, dataset: DatasetModel, session: Session) -> N
...
-        session.execute(
-            stmt,
-            [{'target_dag_id': target_dag.dag_id} for target_dag in dataset.consuming_dags],
-        )
-        session.flush()
+        params = [{'target_dag_id': target_dag.dag_id} for target_dag in dataset.consuming_dags]
+        if params:
+            stmt = insert(DatasetDagRunQueue).values(dataset_id=dataset.id).on_conflict_do_nothing()
+            session.execute(stmt, params)
+            session.flush()
```
Comment (kaxil) [inline airflow/datasets/manager.py]: Change looks good but we should add a test case for it

Comment (ashb) [reply airflow/datasets/manager.py]: It also only cover one path, there are two.

Comment (ashb) [reply airflow/datasets/manager.py]: Updated with test and a simpler place to make the change.

**Review by kaxil — APPROVED** [review]

File: airflow/datasets/manager.py (hunk 1 above)
Comment (dstandish) [reply airflow/datasets/manager.py]: the other path wasn't susceptible to the issue

**Pull request merged**